libraries/AP_ExternalAHRS/AP_ExternalAHRS_AdvancedNavigation.h

```diff
@@ -0,0 +1,378 @@
+        device_id_air_data_unit,
+        device_id_spatial_fog_dual = 16,
+        device_id_motus,
+        device_id_gnss_compass,
+        device_id_certus = 26,
+        device_id_aries,
+        device_id_boreas_d90,
+        device_id_boreas_d90_fpga = 35,
+        device_id_boreas_coil,
+        device_id_certus_mini_a = 49,
+        device_id_certus_mini_n,
+        device_id_certus_mini_d,
+    } device_id_e;
+
+
+
+    struct PACKED AN_ACKNOWLEGE {
+        uint8_t id_acknowledged;
+        uint16_t crc_acknowledged;
+        uint8_t result;
+    };
+
+    struct PACKED AN_DEVICE_INFO {
+        uint32_t software_version;
+        uint32_t device_id;
+        uint32_t hardware_revision;
+        uint32_t serial_1;
+        uint32_t serial_2;
+        uint32_t serial_3;
+    };
+
+    struct PACKED AN_SYSTEM_STATE {
+        union {
+            uint16_t r;
+            struct {
+                uint16_t system_failure :1;
+                uint16_t accelerometer_sensor_failure :1;
+                uint16_t gyroscope_sensor_failure :1;
+                uint16_t magnetometer_sensor_failure :1;
+                uint16_t pressure_sensor_failure :1;
+                uint16_t gnss_failure :1;
+                uint16_t accelerometer_over_range :1;
+                uint16_t gyroscope_over_range :1;
+                uint16_t magnetometer_over_range :1;
+                uint16_t pressure_over_range :1;
+                uint16_t minimum_temperature_alarm :1;
+                uint16_t maximum_temperature_alarm :1;
+                uint16_t internal_data_logging_error :1;
+                uint16_t high_voltage_alarm :1;
+                uint16_t gnss_antenna_fault :1;
+                uint16_t serial_port_overflow_alarm :1;
+            } b;
+        } system_status;
+        union {
+            uint16_t r;
+            struct {
+                uint16_t orientation_filter_initialised :1;
+                uint16_t ins_filter_initialised :1;
+                uint16_t heading_initialised :1;
+                uint16_t utc_time_initialised :1;
+                uint16_t gnss_fix_type :3;
+                uint16_t event1_flag :1;
+                uint16_t event2_flag :1;
+                uint16_t internal_gnss_enabled :1;
+                uint16_t dual_antenna_heading_active :1;
+                uint16_t velocity_heading_enabled :1;
+                uint16_t atmospheric_altitude_enabled :1;
+                uint16_t external_position_active :1;
+                uint16_t external_velocity_active :1;
+                uint16_t external_heading_active :1;
+            } b;
+        } filter_status;
+        uint32_t unix_time_seconds;
+        uint32_t microseconds;
+        double latitude; // rad
+        double longitude;
+        double height; // m
+        float velocity_n; // m/s
+        float velocity_e;
+        float velocity_d;
+        float body_acceleration_x; // m/s/s
+        float body_acceleration_y;
+        float body_acceleration_z;
```

Comment:
```suggestion
        Vector3f body_acceleration; // m/s/s
```

... and accompanying changes

Similarly where we can do it for e.g. vel-ned and angular velocity.